bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdFallbackManager.java

```diff
@@ -68,9 +68,9 @@ class FallbackFlags {
 	public FallbackFlags getFallbackFlagsFromInitialUri(URI root) {
 		if (isWindows) {
 			return uri.find(root)//
-					.flatMap(new UriResource(workspace))//
-					.flatMap(this::flags)//
-					.orElse(null);
+					.flatMap(initUri -> new UriResource(workspace).apply(initUri))//
```

Comment:
These changes look semantically identical, but a bit more verbose. Or am I miss something?

Reply:
Will `apply` on `UriResource` be called in the previous statement? I think that was the problem: Previously the `root` resource has been mapped to `this::flags`, but wee need the `IResource` from `initialFile` which we get from  `uri.find(root)` (btw: find does not need any input arguemt)

Reply:
I have problems to understand what the previous statement does. If they do the same, I think my proposal is easier to understand, because the `apply` call on `UriResource(workspace)` is explicitly stated.

Reply:
I just switched back to old statement and the fallback flags were null. So I assume that the statements are not identical. 

I think that: 
`.flatMap(initUri -> new UriResource(workspace).apply(initUri))` and
`.flatMap(new UriResource(workspace))`
do not the same things

Reply:
I don't understand why we have to care about `IContainer` at all. The URI we use should always point to files.

Reply:
> I think my proposal is easier to understand, because the apply call on UriResource(workspace) is explicitly stated.

`UriResource` class was designed to be used with `Optional` in a short form, and the question is: do we use Java 8 constructs to write code or not.
If we do - short form is preferable.
If we don't - we should be consistent and rework `Optional` to a deep-nested `if-else` tree as well. But in the case we don't I would expect the corresponding notice in the "Contributor Guide"

Reply:
I am fine Java 8 constructs and with the short form, but these two statements obviously behave different and I can't say why.